group-coordinator/src/main/java/org/apache/kafka/coordinator/group/GroupMetadataManager.java

```diff
@@ -1993,6 +2002,7 @@ private CoordinatorResult<StreamsGroupHeartbeatResult, CoordinatorRecord> stream
             response.setActiveTasks(createStreamsGroupHeartbeatResponseTaskIds(updatedMember.assignedTasks().activeTasks()));
             response.setStandbyTasks(createStreamsGroupHeartbeatResponseTaskIds(updatedMember.assignedTasks().standbyTasks()));
             response.setWarmupTasks(createStreamsGroupHeartbeatResponseTaskIds(updatedMember.assignedTasks().warmupTasks()));
+            group.setEndpointInformationEpoch(group.endpointInformationEpoch() + 1);
```

Comment:
Bump the endpoint information epoch here as there's been assignment changes.  The endpoint information will be sent in the next heartbeat response.   We'll do it this way, as the next successful heartbeat request/response cycle the group member will have accepted the new assignment.

Reply:
Do we need to change the reponse, too, using the bumped epoch, as also ensure the endpoint information is set?

Reply:
maybe we can move the block around setting endpoint information in the response below this block.